src/main/java/org/springframework/retry/policy/ExpressionRetryPolicy.java

```diff
@@ -114,4 +119,27 @@ public boolean canRetry(RetryContext context) {
 		}
 	}
 
+	/**
+	 * Check if the expression is a template
+	 * @param expression the expression string
+	 * @return true if the expression string is a template
+	 */
+	private boolean isTemplate(String expression) {
+		return expression.contains(PARSER_CONTEXT.getExpressionPrefix()) && expression.contains(PARSER_CONTEXT.getExpressionSuffix());
+	}
+
+	/**
+	 * Get expression based on the expression string.
+	 * At the moment supports both literal and template expressions. Template expressions are deprecated.
+	 * @param expression the expression string
+	 * @return literal expression or template expression
+	 */
+	private Expression getExpression(String expression) {
+		if (isTemplate(expression)) {
+			logger.warn("Template expressions are deprecated in favor of literal expressions");
```

Comment:
I would say just `Template expressions are deprecated for ExpressionRetryPolicy` .

We really can't say *literal expressions* since that is going to confuse and users just because there is really a `LiteralExpresion` to return just plain string all the time. So, we can't say that here to bring end-users in the situation when they may decide that this is trealy going to be only about simple strings.

Reply:
Perhaps something more descriptive would be more helpful for novice users; something like 

    "#{...} syntax is not required for this run-time expression and is deprecated in favor of a simple expression string"

Reply:
done